### PR TITLE
Add drag and sort functionalities

### PR DESCRIPTION
**Pull request by leonardodiasb**
In this feature I created:
- Two JavaScript files, containing classes for drag drop and sort methods, and storage updates methods.
- dragstart, dragover, drop and change event listeners.

### CODE REVIEW

File: src/drag.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const container = document.querySelector('#td-section');
+
+export default class DragDropSort {
+  dragOver = (e) => {
+    e.preventDefault();
+    const draggable = document.querySelector('.dragging');
+    const draggableElements = [...container.querySelectorAll('.td-item:not(.dragging)')];
+    const afterElement = draggableElements.reduce((closest, child) => {
+      const box = child.getBoundingClientRect();
+      const offset = e.clientY - box.top - box.height / 2;
```
Comment (Davidosky007) [inline src/drag.js:12]: * Kindly fix the checkbox, drag and drop functionality, currently i get errors in the console when i try to drag and drop the list 
* --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
image

File: src/index.js [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
-  },
-];
+const tdListStored = JSON.parse(localStorage.getItem('ToDoList'));
+let tdList = tdListStored;
+if (tdList == null) {
+  tdList = [
+    { description: 'complete To Do List project', completed: true, index: '0' },
+    { description: 'wash the dishes', completed: true, index: '1' },
+    { description: 'rest', completed: true, index: '2' }];
+}
```
Comment (Davidosky007) [inline src/index.js]: * I suggest you update the local storage with the change in the array list caused by the event listeners (for example when a list is dragged or staus changed to a new position the index and status are updated  in the local storage even when the page reloads  ) [refer to the project requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Store%20the%20updated%20array%20of%20items%20in%20local%20storage%2C%20so%20the%20user%20gets%20the%20correct%20list%20values%20after%20the%20page%20reloads.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @leonardodiasb 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
* I have left some comments below please go through them and implement them as they are required for this project 🙏 
## _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js (hunk 2 above)
Comment (DeliceLydia) [reply src/index.js]: Well addressed this!

File: src/drag.js (hunk 1 above)
Comment (DeliceLydia) [reply src/drag.js:12]: Resolved this

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉 🎉 

## Hi @leonardodiasb , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**